File: packages/typescript/bin/executar_testes.ts
import { CriptografiaAlvo } from '../src/Seguranca/CriptografiaAlvo.ts';
import { SuiteDeAtaques } from '../src/Seguranca/SuiteDeAtaques.ts';

import { AtaqueIdaEVolta } from '../src/Seguranca/Ataques/AtaqueIdaEVolta.ts';
import { AtaqueAdulteracao } from '../src/Seguranca/Ataques/AtaqueAdulteracao.ts';
import { AtaqueColisaoIV } from '../src/Seguranca/Ataques/AtaqueColisaoIV.ts';
import { AtaqueBytesFixos } from '../src/Seguranca/Ataques/AtaqueBytesFixos.ts';
import { AtaqueAvalanche } from '../src/Seguranca/Ataques/AtaqueAvalanche.ts';
import { AtaqueDistribuicaoBytes } from '../src/Seguranca/Ataques/AtaqueDistribuicaoBytes.ts';
import { AtaqueFoldEstrutural } from '../src/Seguranca/Ataques/AtaqueFoldEstrutural.ts';
import { AtaqueCoberturaDependencia } from '../src/Seguranca/Ataques/AtaqueCoberturaDependencia.ts';
import { AtaqueIVsDegenerados } from '../src/Seguranca/Ataques/AtaqueIVsDegenerados.ts';
import { AtaqueCorrelacaoMesmoPlaintext } from '../src/Seguranca/Ataques/AtaqueCorrelacaoMesmoPlaintext.ts';
import { AtaqueReusoIV } from '../src/Seguranca/Ataques/AtaqueReusoIV.ts';
import { AtaqueVetorDeterministico } from '../src/Seguranca/Ataques/AtaqueVetorDeterministico.ts';
import { AtaqueTiming } from '../src/Seguranca/Ataques/AtaqueTiming.ts';
import { AtaqueColisaoChecksum } from '../src/Seguranca/Ataques/AtaqueColisaoChecksum.ts';
import { AtaqueEntropiaIV } from '../src/Seguranca/Ataques/AtaqueEntropiaIV.ts';
import { AtaqueAvalancheChave } from '../src/Seguranca/Ataques/AtaqueAvalancheChave.ts';
import { AtaqueAvalancheChecksum } from '../src/Seguranca/Ataques/AtaqueAvalancheChecksum.ts';
import { AtaqueIndependenciaProposito } from '../src/Seguranca/Ataques/AtaqueIndependenciaProposito.ts';
import { AtaqueAutocorrelacao } from '../src/Seguranca/Ataques/AtaqueAutocorrelacao.ts';
import { AtaqueTokensMalformados } from '../src/Seguranca/Ataques/AtaqueTokensMalformados.ts';
import { AtaqueChaveErrada } from '../src/Seguranca/Ataques/AtaqueChaveErrada.ts';
import { AtaqueComplexidadeLinear } from '../src/Seguranca/Ataques/AtaqueComplexidadeLinear.ts';
import { AtaqueBateriaEstatistica } from '../src/Seguranca/Ataques/AtaqueBateriaEstatistica.ts';
import { AtaqueCanonicalizacaoToken } from '../src/Seguranca/Ataques/AtaqueCanonicalizacaoToken.ts';
import { AtaqueCoberturaDependenciaIV } from '../src/Seguranca/Ataques/AtaqueCoberturaDependenciaIV.ts';
import { AtaqueSeparacaoChaveIV } from '../src/Seguranca/Ataques/AtaqueSeparacaoChaveIV.ts';
import { AtaqueChavesDegeneradas } from '../src/Seguranca/Ataques/AtaqueChavesDegeneradas.ts';
import { AtaqueSerialBits } from '../src/Seguranca/Ataques/AtaqueSerialBits.ts';
import { AtaqueCusum } from '../src/Seguranca/Ataques/AtaqueCusum.ts';
import { AtaqueEntropiaAproximada } from '../src/Seguranca/Ataques/AtaqueEntropiaAproximada.ts';
import { AtaqueMensagemLonga } from '../src/Seguranca/Ataques/AtaqueMensagemLonga.ts';
import { AtaqueIdaEVoltaBinario } from '../src/Seguranca/Ataques/AtaqueIdaEVoltaBinario.ts';
import { AtaqueConfusaoCampos } from '../src/Seguranca/Ataques/AtaqueConfusaoCampos.ts';
import { AtaqueLinearidadeChecksum } from '../src/Seguranca/Ataques/AtaqueLinearidadeChecksum.ts';
import { AtaqueValidacaoChave } from '../src/Seguranca/Ataques/AtaqueValidacaoChave.ts';
import { AtaqueIntegral } from '../src/Seguranca/Ataques/AtaqueIntegral.ts';
import { AtaqueDiferencialKeystream } from '../src/Seguranca/Ataques/AtaqueDiferencialKeystream.ts';
import { AtaqueDistribuicaoPorPosicao } from '../src/Seguranca/Ataques/AtaqueDistribuicaoPorPosicao.ts';
import { AtaqueCorrelacaoPosicoes } from '../src/Seguranca/Ataques/AtaqueCorrelacaoPosicoes.ts';
import { AtaquePreditorDeBits } from '../src/Seguranca/Ataques/AtaquePreditorDeBits.ts';

const alvo = new CriptografiaAlvo();

const suite = new SuiteDeAtaques();
suite
  .adicionar(new AtaqueIdaEVolta())
  .adicionar(new AtaqueAdulteracao())
  .adicionar(new AtaqueColisaoIV())
  .adicionar(new AtaqueBytesFixos())
  .adicionar(new AtaqueAvalanche())
  .adicionar(new AtaqueDistribuicaoBytes())
  .adicionar(new AtaqueFoldEstrutural())
  .adicionar(new AtaqueCoberturaDependencia())
  .adicionar(new AtaqueIVsDegenerados())
  .adicionar(new AtaqueCorrelacaoMesmoPlaintext())
  .adicionar(new AtaqueReusoIV())
  .adicionar(new AtaqueVetorDeterministico())
  .adicionar(new AtaqueTiming())
  .adicionar(new AtaqueColisaoChecksum())
  .adicionar(new AtaqueEntropiaIV())
  .adicionar(new AtaqueAvalancheChave())
  .adicionar(new AtaqueAvalancheChecksum())
  .adicionar(new AtaqueIndependenciaProposito())
  .adicionar(new AtaqueAutocorrelacao())
  .adicionar(new AtaqueTokensMalformados())
  .adicionar(new AtaqueChaveErrada())
  .adicionar(new AtaqueComplexidadeLinear())
  .adicionar(new AtaqueBateriaEstatistica())
  .adicionar(new AtaqueCanonicalizacaoToken())
  .adicionar(new AtaqueCoberturaDependenciaIV())
  .adicionar(new AtaqueSeparacaoChaveIV())
  .adicionar(new AtaqueChavesDegeneradas())
  .adicionar(new AtaqueSerialBits())
  .adicionar(new AtaqueCusum())
  .adicionar(new AtaqueEntropiaAproximada())
  .adicionar(new AtaqueMensagemLonga())
  .adicionar(new AtaqueIdaEVoltaBinario())
  .adicionar(new AtaqueConfusaoCampos())
  .adicionar(new AtaqueLinearidadeChecksum())
  .adicionar(new AtaqueValidacaoChave())
  .adicionar(new AtaqueIntegral())
  .adicionar(new AtaqueDiferencialKeystream())
  .adicionar(new AtaqueDistribuicaoPorPosicao())
  .adicionar(new AtaqueCorrelacaoPosicoes())
  .adicionar(new AtaquePreditorDeBits());

const passou = suite.rodarEImprimir(alvo);

process.exit(passou ? 0 : 1);

File: packages/python/bin/executar_testes.py
#!/usr/bin/env python3
"""Roda a suíte de ataques contra a implementação Python da cifra."""

import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from src.Seguranca.CriptografiaAlvo import CriptografiaAlvo
from src.Seguranca.SuiteDeAtaques import SuiteDeAtaques

from src.Seguranca.Ataques.AtaqueIdaEVolta import AtaqueIdaEVolta
from src.Seguranca.Ataques.AtaqueAdulteracao import AtaqueAdulteracao
from src.Seguranca.Ataques.AtaqueColisaoIV import AtaqueColisaoIV
from src.Seguranca.Ataques.AtaqueBytesFixos import AtaqueBytesFixos
from src.Seguranca.Ataques.AtaqueAvalanche import AtaqueAvalanche
from src.Seguranca.Ataques.AtaqueDistribuicaoBytes import AtaqueDistribuicaoBytes
from src.Seguranca.Ataques.AtaqueFoldEstrutural import AtaqueFoldEstrutural
from src.Seguranca.Ataques.AtaqueCoberturaDependencia import AtaqueCoberturaDependencia
from src.Seguranca.Ataques.AtaqueIVsDegenerados import AtaqueIVsDegenerados
from src.Seguranca.Ataques.AtaqueCorrelacaoMesmoPlaintext import AtaqueCorrelacaoMesmoPlaintext
from src.Seguranca.Ataques.AtaqueReusoIV import AtaqueReusoIV
from src.Seguranca.Ataques.AtaqueVetorDeterministico import AtaqueVetorDeterministico
from src.Seguranca.Ataques.AtaqueTiming import AtaqueTiming
from src.Seguranca.Ataques.AtaqueColisaoChecksum import AtaqueColisaoChecksum
from src.Seguranca.Ataques.AtaqueEntropiaIV import AtaqueEntropiaIV
from src.Seguranca.Ataques.AtaqueAvalancheChave import AtaqueAvalancheChave
from src.Seguranca.Ataques.AtaqueAvalancheChecksum import AtaqueAvalancheChecksum
from src.Seguranca.Ataques.AtaqueIndependenciaProposito import AtaqueIndependenciaProposito
from src.Seguranca.Ataques.AtaqueAutocorrelacao import AtaqueAutocorrelacao
from src.Seguranca.Ataques.AtaqueTokensMalformados import AtaqueTokensMalformados
from src.Seguranca.Ataques.AtaqueChaveErrada import AtaqueChaveErrada
from src.Seguranca.Ataques.AtaqueComplexidadeLinear import AtaqueComplexidadeLinear
from src.Seguranca.Ataques.AtaqueBateriaEstatistica import AtaqueBateriaEstatistica
from src.Seguranca.Ataques.AtaqueCanonicalizacaoToken import AtaqueCanonicalizacaoToken
from src.Seguranca.Ataques.AtaqueCoberturaDependenciaIV import AtaqueCoberturaDependenciaIV
from src.Seguranca.Ataques.AtaqueSeparacaoChaveIV import AtaqueSeparacaoChaveIV
from src.Seguranca.Ataques.AtaqueChavesDegeneradas import AtaqueChavesDegeneradas
from src.Seguranca.Ataques.AtaqueSerialBits import AtaqueSerialBits
from src.Seguranca.Ataques.AtaqueCusum import AtaqueCusum
from src.Seguranca.Ataques.AtaqueEntropiaAproximada import AtaqueEntropiaAproximada
from src.Seguranca.Ataques.AtaqueMensagemLonga import AtaqueMensagemLonga
from src.Seguranca.Ataques.AtaqueIdaEVoltaBinario import AtaqueIdaEVoltaBinario
from src.Seguranca.Ataques.AtaqueConfusaoCampos import AtaqueConfusaoCampos
from src.Seguranca.Ataques.AtaqueLinearidadeChecksum import AtaqueLinearidadeChecksum
from src.Seguranca.Ataques.AtaqueValidacaoChave import AtaqueValidacaoChave
from src.Seguranca.Ataques.AtaqueIntegral import AtaqueIntegral
from src.Seguranca.Ataques.AtaqueDiferencialKeystream import AtaqueDiferencialKeystream
from src.Seguranca.Ataques.AtaqueDistribuicaoPorPosicao import AtaqueDistribuicaoPorPosicao
from src.Seguranca.Ataques.AtaqueCorrelacaoPosicoes import AtaqueCorrelacaoPosicoes
from src.Seguranca.Ataques.AtaquePreditorDeBits import AtaquePreditorDeBits


def main() -> int:
    alvo = CriptografiaAlvo()

    suite = SuiteDeAtaques()
    (
        suite.adicionar(AtaqueIdaEVolta())
        .adicionar(AtaqueAdulteracao())
        .adicionar(AtaqueColisaoIV())
        .adicionar(AtaqueBytesFixos())
        .adicionar(AtaqueAvalanche())
        .adicionar(AtaqueDistribuicaoBytes())
        .adicionar(AtaqueFoldEstrutural())
        .adicionar(AtaqueCoberturaDependencia())
        .adicionar(AtaqueIVsDegenerados())
        .adicionar(AtaqueCorrelacaoMesmoPlaintext())
        .adicionar(AtaqueReusoIV())
        .adicionar(AtaqueVetorDeterministico())
        .adicionar(AtaqueTiming())
        .adicionar(AtaqueColisaoChecksum())
        .adicionar(AtaqueEntropiaIV())
        .adicionar(AtaqueAvalancheChave())
        .adicionar(AtaqueAvalancheChecksum())
        .adicionar(AtaqueIndependenciaProposito())
        .adicionar(AtaqueAutocorrelacao())
        .adicionar(AtaqueTokensMalformados())
        .adicionar(AtaqueChaveErrada())
        .adicionar(AtaqueComplexidadeLinear())
        .adicionar(AtaqueBateriaEstatistica())
        .adicionar(AtaqueCanonicalizacaoToken())
        .adicionar(AtaqueCoberturaDependenciaIV())
        .adicionar(AtaqueSeparacaoChaveIV())
        .adicionar(AtaqueChavesDegeneradas())
        .adicionar(AtaqueSerialBits())
        .adicionar(AtaqueCusum())
        .adicionar(AtaqueEntropiaAproximada())
        .adicionar(AtaqueMensagemLonga())
        .adicionar(AtaqueIdaEVoltaBinario())
        .adicionar(AtaqueConfusaoCampos())
        .adicionar(AtaqueLinearidadeChecksum())
        .adicionar(AtaqueValidacaoChave())
        .adicionar(AtaqueIntegral())
        .adicionar(AtaqueDiferencialKeystream())
        .adicionar(AtaqueDistribuicaoPorPosicao())
        .adicionar(AtaqueCorrelacaoPosicoes())
        .adicionar(AtaquePreditorDeBits())
    )

    passou = suite.rodar_e_imprimir(alvo)
    return 0 if passou else 1


if __name__ == "__main__":
    sys.exit(main())
